In [1]:
import json
import pickle
def load_data(file_path):
    """ Load the data from a pickle file. """
    with open(file_path, 'rb') as f:
        return pickle.load(f)

In [7]:
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi_exp/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[126,  79,  27,   5],
       [118, 122,  81,  27],
       [ 15,  24,  60,  33],
       [  4,   8,  22,  25]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.53      0.50       237\n > 3 & ≤ 7 days       0.52      0.35      0.42       348\n> 7 & ≤ 14 days       0.32      0.45      0.37       132\n      > 14 days       0.28      0.42      0.34        59\n\n       accuracy                           0.43       776\n      macro avg       0.40      0.44      0.41       776\n   weighted avg       0.46      0.43      0.43       776\n'}, 'f1': {'test': 0.40805171346103913}}}


# Global KL

In [3]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/global_kl/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[126,  79,  27,   5],
       [118, 122,  81,  27],
       [ 15,  24,  60,  33],
       [  4,   8,  22,  25]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.53      0.50       237\n > 3 & ≤ 7 days       0.52      0.35      0.42       348\n> 7 & ≤ 14 days       0.32      0.45      0.37       132\n      > 14 days       0.28      0.42      0.34        59\n\n       accuracy                           0.43       776\n      macro avg       0.40      0.44      0.41       776\n   weighted avg       0.46      0.43      0.43       776\n'}, 'f1': {'test': 0.40805171346103913}}}


In [5]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/global_kl/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 88, 139,   6,   4],
       [ 78, 236,  24,  10],
       [ 14,  81,  24,  13],
       [  1,  33,  12,  13]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.49      0.37      0.42       237\n > 3 & ≤ 7 days       0.48      0.68      0.56       348\n> 7 & ≤ 14 days       0.36      0.18      0.24       132\n      > 14 days       0.33      0.22      0.26        59\n\n       accuracy                           0.47       776\n      macro avg       0.41      0.36      0.37       776\n   weighted avg       0.45      0.47      0.44       776\n'}, 'f1': {'test': 0.37250547352414926}}}


- missingInd

In [6]:
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/global_kl/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[1375,  296,   29,   29],
       [1262,  742,  210,   87],
       [ 225,  322,  211,   87],
       [  67,  142,  115,   65]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.80      0.59      1729\n > 3 & ≤ 7 days       0.49      0.32      0.39      2301\n> 7 & ≤ 14 days       0.37      0.25      0.30       845\n      > 14 days       0.24      0.17      0.20       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.38      0.37      5264\n   weighted avg       0.45      0.45      0.43      5264\n'}, 'f1': {'test': 0.3694400672820709}}}


# MI Only

In [4]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[126,  79,  27,   5],
       [118, 122,  81,  27],
       [ 15,  24,  60,  33],
       [  4,   8,  22,  25]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.53      0.50       237\n > 3 & ≤ 7 days       0.52      0.35      0.42       348\n> 7 & ≤ 14 days       0.32      0.45      0.37       132\n      > 14 days       0.28      0.42      0.34        59\n\n       accuracy                           0.43       776\n      macro avg       0.40      0.44      0.41       776\n   weighted avg       0.46      0.43      0.43       776\n'}, 'f1': {'test': 0.40805171346103913}}}


In [3]:
# ts-cxr-txt
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 88, 139,   6,   4],
       [ 78, 236,  24,  10],
       [ 14,  81,  24,  13],
       [  1,  33,  12,  13]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.49      0.37      0.42       237\n > 3 & ≤ 7 days       0.48      0.68      0.56       348\n> 7 & ≤ 14 days       0.36      0.18      0.24       132\n      > 14 days       0.33      0.22      0.26        59\n\n       accuracy                           0.47       776\n      macro avg       0.41      0.36      0.37       776\n   weighted avg       0.45      0.47      0.44       776\n'}, 'f1': {'test': 0.37250547352414926}}}


In [4]:
# ts-cxr-ecg
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 99, 119,  19,   0],
       [ 86, 207,  55,   0],
       [ 22,  64,  46,   0],
       [  4,  29,  26,   0]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.42      0.44       237\n > 3 & ≤ 7 days       0.49      0.59      0.54       348\n> 7 & ≤ 14 days       0.32      0.35      0.33       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.32      0.34      0.33       776\n   weighted avg       0.42      0.45      0.43       776\n'}, 'f1': {'test': 0.3281662142347958}}}


In [5]:
# ts-ecg-txt
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[123,  95,  19,   0],
       [118, 156,  74,   0],
       [ 16,  50,  66,   0],
       [  2,  21,  36,   0]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.52      0.50       237\n > 3 & ≤ 7 days       0.48      0.45      0.47       348\n> 7 & ≤ 14 days       0.34      0.50      0.40       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.44       776\n      macro avg       0.32      0.37      0.34       776\n   weighted avg       0.42      0.44      0.43       776\n'}, 'f1': {'test': 0.3413272771242927}}}


- missingInd

In [6]:
# 4 mod
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 898,  763,   49,   19],
       [ 658, 1337,  215,   91],
       [  68,  475,  210,   92],
       [  18,  188,  115,   68]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.55      0.52      0.53      1729\n > 3 & ≤ 7 days       0.48      0.58      0.53      2301\n> 7 & ≤ 14 days       0.36      0.25      0.29       845\n      > 14 days       0.25      0.17      0.21       389\n\n       accuracy                           0.48      5264\n      macro avg       0.41      0.38      0.39      5264\n   weighted avg       0.47      0.48      0.47      5264\n'}, 'f1': {'test': 0.3900202467580542}}}


In [7]:
# ts-cxr-txt
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[1375,  296,   29,   29],
       [1262,  742,  210,   87],
       [ 225,  322,  211,   87],
       [  67,  142,  115,   65]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.80      0.59      1729\n > 3 & ≤ 7 days       0.49      0.32      0.39      2301\n> 7 & ≤ 14 days       0.37      0.25      0.30       845\n      > 14 days       0.24      0.17      0.20       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.38      0.37      5264\n   weighted avg       0.45      0.45      0.43      5264\n'}, 'f1': {'test': 0.3694400672820709}}}


In [8]:
# ts-cxr-ecg
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[1196,  477,   56,    0],
       [1067,  976,  251,    7],
       [ 176,  396,  267,    6],
       [  43,  176,  166,    4]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.69      0.57      1729\n > 3 & ≤ 7 days       0.48      0.42      0.45      2301\n> 7 & ≤ 14 days       0.36      0.32      0.34       845\n      > 14 days       0.24      0.01      0.02       389\n\n       accuracy                           0.46      5264\n      macro avg       0.39      0.36      0.34      5264\n   weighted avg       0.44      0.46      0.44      5264\n'}, 'f1': {'test': 0.34396854926029446}}}


In [2]:
# ts-ecg-txt
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/mi/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[1012,  630,   60,   27],
       [ 873,  975,  337,  116],
       [ 133,  291,  327,   94],
       [  44,  100,  170,   75]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.49      0.59      0.53      1729\n > 3 & ≤ 7 days       0.49      0.42      0.45      2301\n> 7 & ≤ 14 days       0.37      0.39      0.38       845\n      > 14 days       0.24      0.19      0.21       389\n\n       accuracy                           0.45      5264\n      macro avg       0.40      0.40      0.39      5264\n   weighted avg       0.45      0.45      0.45      5264\n'}, 'f1': {'test': 0.3944398210633714}}}


# KL+MI (smooth=0.1)

In [ ]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 969,  690,   38,   32],
       [ 791, 1101,  170,  239],
       [  88,  352,  147,  258],
       [  22,  135,   77,  155]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.52      0.56      0.54      1729\n > 3 & ≤ 7 days       0.48      0.48      0.48      2301\n> 7 & ≤ 14 days       0.34      0.17      0.23       845\n      > 14 days       0.23      0.40      0.29       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.40      0.38      5264\n   weighted avg       0.45      0.45      0.45      5264\n'}, 'f1': {'test': 0.38462765754220835}}}


In [ ]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[   0, 1587,  135,    7],
          [   0, 1776,  494,   31],
          [   0,  428,  394,   23],
          [   0,  141,  225,   23]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.00      0.00      0.00      1729\n > 3 & ≤ 7 days       0.45      0.77      0.57      2301\n> 7 & ≤ 14 days       0.32      0.47      0.38       845\n      > 14 days       0.27      0.06      0.10       389\n\n       accuracy                           0.42      5264\n      macro avg       0.26      0.32      0.26      5264\n   weighted avg       0.27      0.42      0.32      5264\n'},
  'f1': {'test': 0.2609036760738692}}}

In [ ]:
## 4 label TS_ECG_Text -- with smooth init=0.5 decay=0.1
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi_0.1/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 139, 1567,   23,    0],
       [ 293, 1802,  189,   17],
       [ 203,  456,  168,   18],
       [ 128,  162,   85,   14]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.18      0.08      0.11      1729\n > 3 & ≤ 7 days       0.45      0.78      0.57      2301\n> 7 & ≤ 14 days       0.36      0.20      0.26       845\n      > 14 days       0.29      0.04      0.06       389\n\n       accuracy                           0.40      5264\n      macro avg       0.32      0.27      0.25      5264\n   weighted avg       0.34      0.40      0.33      5264\n'}, 'f1': {'test': 0.2512819222215191}}}


In [ ]:
# TS CXR Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[1110,  527,   49,   43],
          [1040,  875,  216,  170],
          [ 192,  282,  223,  148],
          [  43,  132,  120,   94]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.64      0.54      1729\n > 3 & ≤ 7 days       0.48      0.38      0.43      2301\n> 7 & ≤ 14 days       0.37      0.26      0.31       845\n      > 14 days       0.21      0.24      0.22       389\n\n       accuracy                           0.44      5264\n      macro avg       0.38      0.38      0.37      5264\n   weighted avg       0.44      0.44      0.43      5264\n'},
  'f1': {'test': 0.37359688848968786}}}

# KL Only (smooth=0.1)

In [ ]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 921,  714,   47,   47],
       [ 702, 1121,  245,  233],
       [  72,  344,  186,  243],
       [  17,  117,  117,  138]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.54      0.53      0.54      1729\n > 3 & ≤ 7 days       0.49      0.49      0.49      2301\n> 7 & ≤ 14 days       0.31      0.22      0.26       845\n      > 14 days       0.21      0.35      0.26       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.40      0.39      5264\n   weighted avg       0.46      0.45      0.45      5264\n'}, 'f1': {'test': 0.38605233873042644}}}


In [ ]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[   0, 1544,  163,   22],
          [   1, 1675,  554,   71],
          [   0,  405,  375,   65],
          [   0,  135,  221,   33]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.00      0.00      0.00      1729\n > 3 & ≤ 7 days       0.45      0.73      0.55      2301\n> 7 & ≤ 14 days       0.29      0.44      0.35       845\n      > 14 days       0.17      0.08      0.11       389\n\n       accuracy                           0.40      5264\n      macro avg       0.23      0.31      0.25      5264\n   weighted avg       0.25      0.40      0.31      5264\n'},
  'f1': {'test': 0.25353560155478655}}}

In [ ]:
## 4 label TS_ECG_Text -- with smooth init=0.5 decay=0.1
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_0.1/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 123, 1573,   14,   19],
       [ 217, 1855,  145,   84],
       [ 128,  489,  162,   66],
       [  91,  181,   71,   46]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.22      0.07      0.11      1729\n > 3 & ≤ 7 days       0.45      0.81      0.58      2301\n> 7 & ≤ 14 days       0.41      0.19      0.26       845\n      > 14 days       0.21      0.12      0.15       389\n\n       accuracy                           0.42      5264\n      macro avg       0.32      0.30      0.28      5264\n   weighted avg       0.35      0.42      0.34      5264\n'}, 'f1': {'test': 0.2753843658349218}}}


In [ ]:
# TS CXR Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_0.1/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[1131,  521,   58,   19],
          [ 993, 1006,  226,   76],
          [ 175,  352,  249,   69],
          [  45,  151,  147,   46]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.65      0.56      1729\n > 3 & ≤ 7 days       0.50      0.44      0.46      2301\n> 7 & ≤ 14 days       0.37      0.29      0.33       845\n      > 14 days       0.22      0.12      0.15       389\n\n       accuracy                           0.46      5264\n      macro avg       0.39      0.38      0.38      5264\n   weighted avg       0.45      0.46      0.45      5264\n'},
  'f1': {'test': 0.3750172818830372}}}

# KL Only

In [4]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 38, 178,  21,   0],
       [ 34, 240,  70,   4],
       [  3,  67,  58,   4],
       [  0,  24,  32,   3]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.16      0.24       237\n > 3 & ≤ 7 days       0.47      0.69      0.56       348\n> 7 & ≤ 14 days       0.32      0.44      0.37       132\n      > 14 days       0.27      0.05      0.09        59\n\n       accuracy                           0.44       776\n      macro avg       0.39      0.34      0.32       776\n   weighted avg       0.44      0.44      0.40       776\n'}, 'f1': {'test': 0.3150011017373762}}}


In [ ]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 23, 207,   7,   0],
          [ 16, 325,   7,   0],
          [  6, 121,   5,   0],
          [  1,  52,   6,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.50      0.10      0.16       237\n > 3 & ≤ 7 days       0.46      0.93      0.62       348\n> 7 & ≤ 14 days       0.20      0.04      0.06       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.29      0.27      0.21       776\n   weighted avg       0.39      0.45      0.34       776\n'},
  'f1': {'test': 0.21088059693612876}}}

In [ ]:
## 4 label TS_CXR_Text -- with smooth init=0.5 decay=0.05
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 39, 190,   8,   0],
       [ 28, 284,  35,   1],
       [  7,  81,  42,   2],
       [  0,  41,  16,   2]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.16      0.25       237\n > 3 & ≤ 7 days       0.48      0.82      0.60       348\n> 7 & ≤ 14 days       0.42      0.32      0.36       132\n      > 14 days       0.40      0.03      0.06        59\n\n       accuracy                           0.47       776\n      macro avg       0.45      0.33      0.32       776\n   weighted avg       0.48      0.47      0.41       776\n'}, 'f1': {'test': 0.3188784488085913}}}


- MissingInd

4 mod // {'42': {'confusion_matrix': {'test': array([[ 949,  693,   51,   36],
          [ 753, 1099,  189,  260],
          [  82,  325,  160,  278],
          [  21,  113,   87,  168]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.55      0.54      1729\n > 3 & ≤ 7 days       0.49      0.48      0.49      2301\n> 7 & ≤ 14 days       0.33      0.19      0.24       845\n      > 14 days       0.23      0.43      0.30       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.41      0.39      5264\n   weighted avg       0.46      0.45      0.45      5264\n'},
  'f1': {'test': 0.3898733930886193}}}

In [ ]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[   0, 1663,   61,    5],
          [   1, 1968,  273,   59],
          [   1,  594,  204,   46],
          [   0,  232,  131,   26]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.00      0.00      0.00      1729\n > 3 & ≤ 7 days       0.44      0.86      0.58      2301\n> 7 & ≤ 14 days       0.30      0.24      0.27       845\n      > 14 days       0.19      0.07      0.10       389\n\n       accuracy                           0.42      5264\n      macro avg       0.23      0.29      0.24      5264\n   weighted avg       0.26      0.42      0.31      5264\n'},
  'f1': {'test': 0.23773831551712107}}}

In [ ]:
# TS CXR TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[1104,  552,   62,   11],
          [1000, 1001,  219,   81],
          [ 177,  357,  242,   69],
          [  48,  153,  142,   46]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.64      0.54      1729\n > 3 & ≤ 7 days       0.49      0.44      0.46      2301\n> 7 & ≤ 14 days       0.36      0.29      0.32       845\n      > 14 days       0.22      0.12      0.15       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.37      0.37      5264\n   weighted avg       0.44      0.45      0.44      5264\n'},
  'f1': {'test': 0.3694390134642215}}}

In [ ]:
# TS ECG Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 131, 1582,   16,    0],
          [ 233, 1891,  160,   17],
          [ 163,  509,  156,   17],
          [ 113,  183,   88,    5]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.20      0.08      0.11      1729\n > 3 & ≤ 7 days       0.45      0.82      0.58      2301\n> 7 & ≤ 14 days       0.37      0.18      0.25       845\n      > 14 days       0.13      0.01      0.02       389\n\n       accuracy                           0.41      5264\n      macro avg       0.29      0.27      0.24      5264\n   weighted avg       0.33      0.41      0.33      5264\n'},
  'f1': {'test': 0.24137641260179304}}}

# KL + MI

In [ ]:
# 4mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[125,  93,  18,   1],
          [118, 170,  47,  13],
          [ 20,  50,  47,  15],
          [  2,  24,  24,   9]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.53      0.50       237\n > 3 & ≤ 7 days       0.50      0.49      0.50       348\n> 7 & ≤ 14 days       0.35      0.36      0.35       132\n      > 14 days       0.24      0.15      0.19        59\n\n       accuracy                           0.45       776\n      macro avg       0.39      0.38      0.38       776\n   weighted avg       0.45      0.45      0.45       776\n'},
  'f1': {'test': 0.38266790301424713}}}

In [ ]:
# TS CXR TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 46, 160,  27,   4],
          [ 37, 207,  89,  15],
          [  4,  49,  68,  11],
          [  1,  17,  30,  11]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.52      0.19      0.28       237\n > 3 & ≤ 7 days       0.48      0.59      0.53       348\n> 7 & ≤ 14 days       0.32      0.52      0.39       132\n      > 14 days       0.27      0.19      0.22        59\n\n       accuracy                           0.43       776\n      macro avg       0.40      0.37      0.36       776\n   weighted avg       0.45      0.43      0.41       776\n'},
  'f1': {'test': 0.3565575338932825}}}

In [ ]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 33, 203,   1,   0],
          [ 27, 321,   0,   0],
          [  6, 125,   1,   0],
          [  1,  56,   2,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.49      0.14      0.22       237\n > 3 & ≤ 7 days       0.46      0.92      0.61       348\n> 7 & ≤ 14 days       0.25      0.01      0.01       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.46       776\n      macro avg       0.30      0.27      0.21       776\n   weighted avg       0.40      0.46      0.34       776\n'},
  'f1': {'test': 0.2103744387993614}}}

In [ ]:
# TS ECG TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 14, 200,  23,   0],
          [ 12, 255,  76,   5],
          [  2,  56,  59,  15],
          [  0,  19,  33,   7]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.50      0.06      0.11       237\n > 3 & ≤ 7 days       0.48      0.73      0.58       348\n> 7 & ≤ 14 days       0.31      0.45      0.37       132\n      > 14 days       0.26      0.12      0.16        59\n\n       accuracy                           0.43       776\n      macro avg       0.39      0.34      0.30       776\n   weighted avg       0.44      0.43      0.37       776\n'},
  'f1': {'test': 0.3036604390192343}}}

- MissingInd

In [ ]:
# 4 mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl") 

{'42': {'confusion_matrix': {'test': array([[ 756,  879,   32,   62],
          [ 622, 1231,  134,  314],
          [  82,  350,  119,  294],
          [  22,  122,   76,  169]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.44      0.47      1729\n > 3 & ≤ 7 days       0.48      0.53      0.50      2301\n> 7 & ≤ 14 days       0.33      0.14      0.20       845\n      > 14 days       0.20      0.43      0.28       389\n\n       accuracy                           0.43      5264\n      macro avg       0.38      0.39      0.36      5264\n   weighted avg       0.44      0.43      0.43      5264\n'},
  'f1': {'test': 0.36191762104221126}}}

In [ ]:
# ts-cxr-text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 930,  738,   55,    6],
          [ 777, 1186,  330,    8],
          [  97,  406,  339,    3],
          [  26,  171,  189,    3]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.54      0.52      1729\n > 3 & ≤ 7 days       0.47      0.52      0.49      2301\n> 7 & ≤ 14 days       0.37      0.40      0.39       845\n      > 14 days       0.15      0.01      0.01       389\n\n       accuracy                           0.47      5264\n      macro avg       0.38      0.37      0.35      5264\n   weighted avg       0.44      0.47      0.45      5264\n'},
  'f1': {'test': 0.3542287546070625}}}

In [ ]:
# ts-ecg-text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 317, 1319,   55,   38],
          [ 279, 1623,  329,   70],
          [  66,  412,  317,   50],
          [  36,  168,  159,   26]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.45      0.18      0.26      1729\n > 3 & ≤ 7 days       0.46      0.71      0.56      2301\n> 7 & ≤ 14 days       0.37      0.38      0.37       845\n      > 14 days       0.14      0.07      0.09       389\n\n       accuracy                           0.43      5264\n      macro avg       0.36      0.33      0.32      5264\n   weighted avg       0.42      0.43      0.40      5264\n'},
  'f1': {'test': 0.3203176032814083}}}

In [ ]:
# ts-cxr-ecg
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 388, 1243,   97,    1],
          [ 250, 1598,  445,    8],
          [  28,  441,  364,   12],
          [   9,  147,  222,   11]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.57      0.22      0.32      1729\n > 3 & ≤ 7 days       0.47      0.69      0.56      2301\n> 7 & ≤ 14 days       0.32      0.43      0.37       845\n      > 14 days       0.34      0.03      0.05       389\n\n       accuracy                           0.45      5264\n      macro avg       0.43      0.34      0.33      5264\n   weighted avg       0.47      0.45      0.41      5264\n'},
  'f1': {'test': 0.3254498157758426}}}

# No Weights New

In [4]:
# 4mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[103,  99,  32,   3],
          [ 93, 135, 103,  17],
          [ 12,  27,  72,  21],
          [  2,   9,  36,  12]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.49      0.43      0.46       237\n > 3 & ≤ 7 days       0.50      0.39      0.44       348\n> 7 & ≤ 14 days       0.30      0.55      0.38       132\n      > 14 days       0.23      0.20      0.21        59\n\n       accuracy                           0.41       776\n      macro avg       0.38      0.39      0.37       776\n   weighted avg       0.44      0.41      0.42       776\n'},
  'f1': {'test': 0.3740072575065082}}}

In [4]:
# TS ECG Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[125,  94,  18,   0],
          [115, 175,  58,   0],
          [ 25,  58,  49,   0],
          [  3,  27,  29,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.53      0.50       237\n > 3 & ≤ 7 days       0.49      0.50      0.50       348\n> 7 & ≤ 14 days       0.32      0.37      0.34       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.32      0.35      0.33       776\n   weighted avg       0.42      0.45      0.43       776\n'},
  'f1': {'test': 0.33407058654583405}}}

In [5]:
# TS CXR Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 75, 148,  13,   1],
          [ 67, 241,  33,   7],
          [ 13,  74,  38,   7],
          [  2,  29,  20,   8]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.32      0.38       237\n > 3 & ≤ 7 days       0.49      0.69      0.57       348\n> 7 & ≤ 14 days       0.37      0.29      0.32       132\n      > 14 days       0.35      0.14      0.20        59\n\n       accuracy                           0.47       776\n      macro avg       0.42      0.36      0.37       776\n   weighted avg       0.45      0.47      0.44       776\n'},
  'f1': {'test': 0.3679190083081495}}}

In [6]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 91, 130,  16,   0],
          [ 77, 227,  44,   0],
          [ 23,  67,  42,   0],
          [  4,  31,  24,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.38      0.42       237\n > 3 & ≤ 7 days       0.50      0.65      0.57       348\n> 7 & ≤ 14 days       0.33      0.32      0.33       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.46       776\n      macro avg       0.32      0.34      0.33       776\n   weighted avg       0.42      0.46      0.44       776\n'},
  'f1': {'test': 0.3280643793247329}}}

- missingInd

In [7]:
# 4 mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[1149,  479,   42,   59],
          [1041,  755,  213,  292],
          [ 182,  210,  184,  269],
          [  49,   82,   94,  164]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.66      0.55      1729\n > 3 & ≤ 7 days       0.49      0.33      0.39      2301\n> 7 & ≤ 14 days       0.35      0.22      0.27       845\n      > 14 days       0.21      0.42      0.28       389\n\n       accuracy                           0.43      5264\n      macro avg       0.38      0.41      0.37      5264\n   weighted avg       0.44      0.43      0.42      5264\n'},
  'f1': {'test': 0.37374461689470245}}}

In [8]:
# TS CXR Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 746,  925,   22,   36],
          [ 613, 1453,   90,  145],
          [  80,  521,  120,  124],
          [  21,  216,   58,   94]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.43      0.47      1729\n > 3 & ≤ 7 days       0.47      0.63      0.54      2301\n> 7 & ≤ 14 days       0.41      0.14      0.21       845\n      > 14 days       0.24      0.24      0.24       389\n\n       accuracy                           0.46      5264\n      macro avg       0.41      0.36      0.36      5264\n   weighted avg       0.46      0.46      0.44      5264\n'},
  'f1': {'test': 0.3636122582793197}}}

In [9]:
# ts-cxr-ecg
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 683,  995,   36,   15],
          [ 510, 1528,  200,   63],
          [  66,  519,  185,   75],
          [  21,  208,  104,   56]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.40      0.45      1729\n > 3 & ≤ 7 days       0.47      0.66      0.55      2301\n> 7 & ≤ 14 days       0.35      0.22      0.27       845\n      > 14 days       0.27      0.14      0.19       389\n\n       accuracy                           0.47      5264\n      macro avg       0.41      0.36      0.37      5264\n   weighted avg       0.46      0.47      0.45      5264\n'},
  'f1': {'test': 0.3654667043634595}}}

In [10]:
# ts-ecg-txt-
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/baseline/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 995,  646,   72,   16],
          [ 810, 1056,  335,  100],
          [ 105,  317,  311,  112],
          [  31,  114,  171,   73]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.58      0.54      1729\n > 3 & ≤ 7 days       0.50      0.46      0.48      2301\n> 7 & ≤ 14 days       0.35      0.37      0.36       845\n      > 14 days       0.24      0.19      0.21       389\n\n       accuracy                           0.46      5264\n      macro avg       0.40      0.40      0.40      5264\n   weighted avg       0.46      0.46      0.46      5264\n'},
  'f1': {'test': 0.397214018738812}}}